joblib/my_exceptions.py

```diff
@@ -64,10 +62,10 @@ def _mk_exception(exception, name=None):
             # We cannot create a subclass: we are already a trivial
             # subclass
             return JoblibException, this_name
-        this_exception = type(this_name, (exception, JoblibException),
-                    dict(__repr__=JoblibException.__repr__,
-                        __str__=JoblibException.__str__),
-                    )
+        elif issubclass(exception, JoblibException):
+            return JoblibException, JoblibException.__name__
+        this_exception = type(
+            this_name, (JoblibException, exception), {})
```

Comment:
The main reason to have exception is to be able to change code from serial to parallel without changing `except` statements:

```python
try:
     for filename in range(filenames):
         process(filename)
except IOError as e:
    handle(e)
```

to:

```python
try:
     Parallel(n_jobs=4)(delayed(process)(filename) for filename in filenames)
except IOError as e:
    handle(e)
```

the handling will still work as `JoblibIOError` is a transportable subclass of `IOError`.